Doc/library/logging.rst

```diff
@@ -1073,89 +1112,30 @@ functions.
 
 .. function:: debug(msg, *args, **kwargs)
 
-   Logs a message with level :const:`DEBUG` on the root logger. The *msg* is the
-   message format string, and the *args* are the arguments which are merged into
-   *msg* using the string formatting operator. (Note that this means that you can
-   use keywords in the format string, together with a single dictionary argument.)
-
-   There are three keyword arguments in *kwargs* which are inspected: *exc_info*
-   which, if it does not evaluate as false, causes exception information to be
-   added to the logging message. If an exception tuple (in the format returned by
-   :func:`sys.exc_info`) or an exception instance is provided, it is used;
-   otherwise, :func:`sys.exc_info` is called to get the exception information.
-
-   The second optional keyword argument is *stack_info*, which defaults to
-   ``False``. If true, stack information is added to the logging
-   message, including the actual logging call. Note that this is not the same
-   stack information as that displayed through specifying *exc_info*: The
-   former is stack frames from the bottom of the stack up to the logging call
-   in the current thread, whereas the latter is information about stack frames
-   which have been unwound, following an exception, while searching for
-   exception handlers.
-
-   You can specify *stack_info* independently of *exc_info*, e.g. to just show
-   how you got to a certain point in your code, even when no exceptions were
-   raised. The stack frames are printed following a header line which says:
+   This is a convenience function that calls ``Logger.debug``, using the root
+   logger as the Logger. The handling of the arguments is in every way identical
+   to what is described in that function.
 
-   .. code-block:: none
+   The only difference is that if the root logger has no handlers, then
+   :func:`basicConfig` is called, prior to calling ``debug`` on the root logger.
 
-       Stack (most recent call last):
+   For very short scripts or quick demonstrations of ``logging`` facilities, ``debug``
+   and the other module level functions may be convenient. However, most programs
+   will want to carefully and explicitly control the setup of the root logger, and
+   should prefer creating a module-level logger and calling ``Logger.debug`` on it,
+   as described at the beginnning of this documentation.
 
-   This mimics the ``Traceback (most recent call last):`` which is used when
```

Comment:
Why has this section been removed?

Reply:
I'm not sure I understand. The git diff is a bit all over the place. "This mimics the traceback..." is still present in the new version, in the Logger.debug section. In the logging.debug section, yes, it's gone, as with almost everything else, since it now just references Logger.debug.